test/jasmine/tests/lib_test.js

```diff
@@ -541,49 +568,49 @@ describe('Test lib.js:', function() {
         it('expands bracketed array notation', function() {
             var input = {'marker[1]': {color: 'red'}};
             var expected = {marker: [undefined, {color: 'red'}]};
-            expect(Lib.expandObjectPaths(input)).toEqual(expected);
+            expectLooseDeepEqual(Lib.expandObjectPaths(input), expected);
         });
 
         it('expands nested arrays', function() {
             var input = {'marker[1].range[1]': 5};
             var expected = {marker: [undefined, {range: [undefined, 5]}]};
             var computed = Lib.expandObjectPaths(input);
-            expect(computed).toEqual(expected);
+            expectLooseDeepEqual(computed, expected);
         });
 
         it('expands bracketed array with more nested attributes', function() {
             var input = {'marker[1]': {'color.alpha': 2}};
             var expected = {marker: [undefined, {color: {alpha: 2}}]};
             var computed = Lib.expandObjectPaths(input);
-            expect(computed).toEqual(expected);
+            expectLooseDeepEqual(computed, expected);
         });
 
         it('expands bracketed array notation without further nesting', function() {
             var input = {'marker[1]': 8};
             var expected = {marker: [undefined, 8]};
             var computed = Lib.expandObjectPaths(input);
-            expect(computed).toEqual(expected);
+            expectLooseDeepEqual(computed, expected);
         });
 
         it('expands bracketed array notation with further nesting', function() {
             var input = {'marker[1].size': 8};
             var expected = {marker: [undefined, {size: 8}]};
             var computed = Lib.expandObjectPaths(input);
-            expect(computed).toEqual(expected);
+            expectLooseDeepEqual(computed, expected);
         });
 
         it('expands bracketed array notation with further nesting', function() {
             var input = {'marker[1].size.magnitude': 8};
             var expected = {marker: [undefined, {size: {magnitude: 8}}]};
             var computed = Lib.expandObjectPaths(input);
-            expect(computed).toEqual(expected);
+            expectLooseDeepEqual(computed, expected);
```

Comment:
Instead of that custom assert routine, why not just have to `expect` statements:

```js
expect(computed.marker[0]).toBeUndefined();
expect(computed.marker[1]).toEqual({ size: { magnitude: 8 } });
```